From: Al Viro <viro@zeniv.linux.org.uk>
Date: Wed, 23 Nov 2011 11:57:51 -0500
Patch-mainline: 3.3
Git-commit: a561be7100cd610bd2e082f3211c1dfb45835817
Subject: [PATCH] switch a bunch of places to mnt_want_write_file()

it's both faster (in case when file has been opened for write) and cleaner.

[Removed non-btrfs filesystems; reason: reduce conflicts in backported patches]

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -206,7 +206,7 @@ static int btrfs_ioctl_setflags(struct f
 		}
 	}
 
-	ret = mnt_want_write(file->f_path.mnt);
+	ret = mnt_want_write_file(file);
 	if (ret)
 		goto out_unlock;
 
@@ -1882,7 +1882,7 @@ static noinline int btrfs_ioctl_snap_des
 		goto out;
 	}
 
-	err = mnt_want_write(file->f_path.mnt);
+	err = mnt_want_write_file(file);
 	if (err)
 		goto out;
 
@@ -2014,7 +2014,7 @@ static int btrfs_ioctl_defrag(struct fil
 	if (btrfs_root_readonly(root))
 		return -EROFS;
 
-	ret = mnt_want_write(file->f_path.mnt);
+	ret = mnt_want_write_file(file);
 	if (ret)
 		return ret;
 
@@ -2244,7 +2244,7 @@ static noinline long btrfs_ioctl_clone(s
 	if (btrfs_root_readonly(root))
 		return -EROFS;
 
-	ret = mnt_want_write(file->f_path.mnt);
+	ret = mnt_want_write_file(file);
 	if (ret)
 		return ret;
 
@@ -2599,7 +2599,7 @@ static long btrfs_ioctl_trans_start(stru
 	if (btrfs_root_readonly(root))
 		goto out;
 
-	ret = mnt_want_write(file->f_path.mnt);
+	ret = mnt_want_write_file(file);
 	if (ret)
 		goto out;
 
